### PR TITLE
Fix Code Scanning Alert tool field

### PR DESCRIPTION
**Pull request by ganeshkumarsv**
The `tool` field on the API response of listing code scanning alerts is a struct instead of a string. This PR fixes the alert tool field to a struct.
Github API: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository

**Deprecation notice from Github**: The tool_name field is deprecated and will, in future, not be included in the response for this endpoint. The example response reflects this change. The tool name can now be found inside the tool field.
https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-analyses-for-a-repository

### CODE REVIEW

**Comment by ganeshkumarsv** [comment]
@googlebot I signed it!

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:21]: While you are touching this file, could you please add a struct comment for both `Tool` and `Alert` describing what they represent?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @ganeshkumarsv !

File: github/code-scanning.go
Comment (stephengroat) [inline github/code-scanning.go:21]:
```suggestion

// Tool represents the tool used to generate a Github Code Scanning Alert
//
// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository
type Tool struct {
```

File: github/code-scanning.go
Comment (stephengroat) [inline github/code-scanning.go:27]:
```suggestion
// Alert represents an individual Github Code Scanning Alert on a single repository
//
// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-alerts-for-a-repository
type Alert struct {
```

**Review by stephengroat — COMMENTED** [review]
these might help

File: github/code-scanning.go
Comment (gmlewis) [reply github/code-scanning.go:21]: Thank you, @stephengroat !

Following idiomatic Go style, please end line 21 with a period to make it a complete sentence for the auto-generated godocs.  The same goes for line 27 below.

File: github/code-scanning.go
Comment (gmlewis) [reply github/code-scanning.go:27]: In addition to the ending period, please spell "GitHub" with a capital `G` and capital `H`.  Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @ganeshkumarsv !
LGTM.

Awaiting second LGTM before merging.

**Comment by gmlewis** [comment]
I apologize for the PR name changes. I see now that this is a "fix" and not an addition... and is therefore a breaking API change.

**Comment by gmlewis** [comment]
Friendly ping @wesleimp

**Comment by ganeshkumarsv** [comment]
@gmlewis @wesleimp 
gentle bump

**Comment by ganeshkumarsv** [comment]
@gmlewis @wesleimp
Can we get one more review for the PR please? We are not able to use `go-github` because of the issue in Alert tool field.

**Comment by gmlewis** [comment]
> @gmlewis @wesleimp
> Can we get one more review for the PR please? We are not able to use `go-github` because of the issue in Alert tool field.

Please feel free to have any contributor to this repo provide a code review and approval, and we will get this merged.  Thanks.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
